matlab/lib/+Ice/+SSL/ConnectionInfo.m

```diff
@@ -1,28 +1,21 @@
-classdef ConnectionInfo < Ice.ConnectionInfo
+classdef (Sealed) ConnectionInfo < Ice.ConnectionInfo
     % ConnectionInfo   Summary of ConnectionInfo
     %
-    % Provides access to the connection detaisl of an SSL connection.
+    % Provides access to the connection details of an SSL connection.
     %
     % ConnectionInfo Properties:
     %   peerCertificate - The peer certificate.
 
     % Copyright (c) ZeroC, Inc. All rights reserved.
 
     methods
-        function obj = ConnectionInfo(underlying, incoming, adapterName, connectionId, ...
-                                      peerCertificate)
-            if nargin == 0
-                underlying = [];
-                incoming = false;
-                adapterName = '';
-                connectionId = '';
-                peerCertificate = '';
-            end
-            obj@Ice.ConnectionInfo(underlying, incoming, adapterName, connectionId);
+        function obj = ConnectionInfo(underlying, peerCertificate)
+            assert(nargin == 2, 'Invalid number of arguments');
```

Comment:
I don't think that this is correct.
MATLAB does not support multiple constructors, there can only be one constructor per class.
And that constructor must always support being called with "default constructor" semantics
(i.e. it must specifically support being called with 0 arguments).

See: https://www.mathworks.com/help/matlab/matlab_oop/class-constructor-methods.html#brd2m9e-3
> If you create a class constructor, ensure it can be called with no input arguments.

----

Note that if you do not provide a custom constructor (as we do here), MATLAB will implicitly 'generate' a default constructor for you.

Reply:
I think what we should do instead is keep the old `if nargin == 0` logic,
but add an `else` block with the `assert nargin == N` in it.

This would ensure that the constructor can only be called with `0` arguments (default constructor) or with all `N` arguments (one-short constructor), but cannot be called with only a subset.

Reply:
Since all these Info classes are created only from the C++ code, I don't see how a default constructor would ever be called.